BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/FileDiagnosticsTelemetryModule.cs

```diff
@@ -171,51 +183,25 @@ private bool SetAndValidateLogsFolder(string filePath, string fileName)
-            }
-            catch (ArgumentException exp)
-            {
-                // Path does not specify a valid file path or contains invalid DirectoryInfo characters.
-                CoreEventSource.Log.LogStorageAccessDeniedError(
-                    GetPathAccessFailureErrorMessage(exp, this.logFilePath, this.logFileName),
-                    LazyInitializer.EnsureInitialized(ref this.windowsIdentityName, this.GetCurrentIdentityName));
-            }
-            catch (DirectoryNotFoundException exp)
+            catch (Exception ex)
             {
-                // The specified path is invalid, such as being on an unmapped drive.
-                CoreEventSource.Log.LogStorageAccessDeniedError(
-                   GetPathAccessFailureErrorMessage(exp, this.logFilePath, this.logFileName),
-                   LazyInitializer.EnsureInitialized(ref this.windowsIdentityName, this.GetCurrentIdentityName));
-            }
-            catch (IOException exp)
-            {
-                // The subdirectory cannot be created. -or- A file or directory already has the name specified by path. -or-  The specified path, file name, or both exceed the system-defined maximum length. .
-                CoreEventSource.Log.LogStorageAccessDeniedError(
-                   GetPathAccessFailureErrorMessage(exp, this.logFilePath, this.logFileName),
-                   LazyInitializer.EnsureInitialized(ref this.windowsIdentityName, this.GetCurrentIdentityName));
-            }
-            catch (SecurityException exp)
-            {
-                // The caller does not have code access permission to create the directory.
+                // NotSupportedException: The given path's format is not supported
+                // UnauthorizedAccessException
+                // ArgumentException: // Path does not specify a valid file path or contains invalid DirectoryInfo characters.
+                // DirectoryNotFoundException: The specified path is invalid, such as being on an unmapped drive.
+                // IOException: The subdirectory cannot be created. -or- A file or directory already has the name specified by path. -or-  The specified path, file name, or both exceed the system-defined maximum length.
+                // SecurityException: The caller does not have code access permission to create the directory.
+
                 CoreEventSource.Log.LogStorageAccessDeniedError(
-                    GetPathAccessFailureErrorMessage(exp, this.logFilePath, this.logFileName),
-                    LazyInitializer.EnsureInitialized(ref this.windowsIdentityName, this.GetCurrentIdentityName));
+                    error: Invariant($"Path: {this.logFilePath} File: {this.logFileName}; Error: {ex.Message}{Environment.NewLine}"),
+                    user: LazyInitializer.EnsureInitialized(ref this.windowsIdentityName, this.GetCurrentIdentityName));
```

Comment:
`GetCurrentIdentityName` is not written for linux, and will throw in non-windows. Can you fix that as well? One of the main target usecase for this feature is non-windows, where perfview is not available.